# Importing pandas to help in reading the uploaded file.

In [50]:
# import pandas
import pandas as pd 

In [51]:
aviation_data =pd.read_csv('aviation-accident-data-2023-05-16.csv')
aviation_data

date                       type registration  \
0        date unk.             Antonov An-12B       T-1206   
1        date unk.             Antonov An-12B       T-1204   
2        date unk.             Antonov An-12B       T-1201   
3        date unk.            Antonov An-12BK          NaN   
4        date unk.            Antonov An-12BP   CCCP-11815   
...            ...                        ...          ...   
23962  11-MAY-2023               Hawker 900XP       PK-LRU   
23963  11-MAY-2023  Cessna 208B Grand Caravan       PK-NGA   
23964  12-MAY-2023  Cessna 208B Grand Caravan       5X-RBR   
23965  14-MAY-2023            Boeing 747-4R7F       LX-OCV   
23966  15-MAY-2023                Learjet 35A       D-CGFQ   

                     operator fatalities             location  \
0               Indonesian AF        NaN                  NaN   
1               Indonesian AF        NaN                  NaN   
2               Indonesian AF        NaN                  NaN   
3                   Soviet AF        NaN  Tiksi Airport (IKS)   
4                   Soviet AF          0  Massawa Airport ...   
...                       ...        ...                  ...   
23962  Angkasa Super Services          0  Maleo Airport (MOH)   
23963  Nasional Global Aviasi          0    Fentheik Airstrip   
23964            Bar Aviation          0  Kampala-Kajjansi...   
23965                Cargolux          0  Luxembourg-Finde...   
23966                     GFD          2        Hohn Air Base   

               country cat     year  
0      Unknown country  U1  unknown  
1      Unknown country  U1  unknown  
2      Unknown country  U1  unknown  
3               Russia  A1  unknown  
4              Eritrea  A1  unknown  
...                ...  ..      ...  
23962        Indonesia  A2     2023  
23963        Indonesia  A2     2023  
23964           Uganda  A1     2023  
23965       Luxembourg  A2     2023  
23966          Germany  A1     2023  

[23967 rows x 9 columns]

PROJECT INTRODUCTION
# Analysis of the Aviation accident dataset
I used the aviation accident dataset provided to help i identifying trends and patterns in aviation safety. The following business problems were looked into:
        1. How have accident occurrences and fatalities changed across decades
        2. Which geographic locations/operators reported the most incidences
        3. Are the modern incidences(accidents) that occurred post 1970 less fatal than the earlier years



# Overview of the Dataset
This dataset contains 23967 rows and 9 columns that entails the following data information
        Date: date of the accident
        Type: the model/type of the aircraft involved
        Registration: the registration number of the aircraft
        Operator: the Airline or military operating the aircraft
        Fatalities: Number of casualties
        country & Location: the geographical location
        Year: year of accident
        

# Data cleaning process
Normalizing missing values(Unknowns, NaN, blanks),
Parse "date" into a standard "datetime" column,
Extracting 'year' and 'decade',
Eliminating duplicates.

In [52]:
# missing values count per column
aviation_data= aviation_data.isna().sum()
aviation_data

date               0
type               0
registration    1548
operator           4
fatalities      3938
location         948
country            0
cat                0
year               0
dtype: int64

In [53]:
#data parsing and Extraction
# converting 'date' column to 'datetime
aviation_data['date_parsed']=pd.to_datetime(aviation_data['date'], errors='coerce')
aviation_data

date                              0
type                              0
registration                   1548
operator                          4
fatalities                     3938
location                        948
country                           0
cat                               0
year                              0
date_parsed     1970-01-01 00:00:00
dtype: object

In [54]:
# Extraction of 'year' and 'decade'
aviation_data['year_extracted']=aviation_data['date_parsed'].year
aviation_data['decade']=(aviation_data['year_extracted']//10)*10
aviation_data

date                                0
type                                0
registration                     1548
operator                            4
fatalities                       3938
location                          948
country                             0
cat                                 0
year                                0
date_parsed       1970-01-01 00:00:00
year_extracted                   1970
decade                           1970
dtype: object

# Exploratory Data Analysis
In this section we look into the accident counts, patterns and trends over the years.The number of accidents recorded per year, the country with the highest number of accident cases.The average fatalities per decade.
We shall use pandas and matplptlib for statistical summaries and visualization.

In [55]:
import matplotlib.pyplot as plt

In [58]:
# Group accidents by year
accident_by_year = aviation_data.groupby("year").size()

# Plot accidents per year
plt.figure(figsize=(12,6))
accident_by_year.plot(kind="line")
plt.title("Number of aviation accidents by year")
plt.xlabel("Year")
plt.ylabel("Number of Accidents")
plt.grid(True)
plt.show()


TypeError: 'int' object is not callable

In [59]:
plt.figure

<function matplotlib.pyplot.figure(num=None, figsize=None, dpi=None, facecolor=None, edgecolor=None, frameon=True, FigureClass=<class 'matplotlib.figure.Figure'>, clear=False, **kwargs)>

In [61]:
accident_by_year = aviation_data.groupby("year").size()

plt.figure(figsize=(12,6))
accident_by_year.plot(kind="line")
plt.title("Number of aviation accidents by year")
plt.xlabel("Year")
plt.ylabel("Number of Accidents")
plt.grid(True)
plt.show()

TypeError: 'int' object is not callable